lib/node/rpc.js

```diff
@@ -2380,6 +2382,30 @@ class RPC extends RPCBase {
     }
   }
 
+  async getNameDNS(args, help) {
+    if (help || args.length < 1 || args.length > 2)
+      throw new RPCError(errs.MISC_ERROR, 'getnamedns "name" "type"');
+
+    const valid = new Validator(args);
+    const name = valid.str(0, '');
+    const value = valid.str(1, 'A');
+
+    if (!name)
+      throw new RPCError(errs.TYPE_ERROR, 'Invalid name.');
+
+    const type = bns.wire.types[value.toUpperCase()];
+
+    if (!type)
+      throw new RPCError(errs.TYPE_ERROR, 'Invalid type.');
+
+    try {
+      const res = await this.node.rs.lookup(bns.util.fqdn(name), type);
```

Comment:
If we merge https://github.com/handshake-org/hsd/pull/554 then endpoints like this will need to check:
 `if (this.node.rs) { ... }` the same way that rpc calls frequently check `if (!this.mempool)` to throw an error if called on an SPV node.